typescript/infra/src/deployment/deploy.ts

```diff
@@ -56,33 +132,108 @@ export async function deployWithArtifacts<Config extends object>({
   // deployments exceeding the timeout are still written
   process.on('beforeExit', handleExit);
 
-  // Deploy the contracts
-  try {
-    await deployer.deploy(targetConfigMap);
-  } catch (e: any) {
-    if (e?.message.includes('Timed out')) {
-      console.warn('Contract deployment exceeding configured timeout', e);
-    } else {
-      console.error('Contract deployment failed', e);
+  // Standard deploy modules are the ones that can be deployed with the
+  // abstract HyperlaneDeployer's deploy function because they don't require any special logic
+  if (standardDeployModules.includes(module)) {
+    await baseDeploy(
+      targetConfigMap,
+      deployer,
+      multiProvider,
+      concurrentDeploy,
+    );
+  } else {
+    try {
+      await deployer.deploy(targetConfigMap);
+    } catch (error: any) {
+      if (error?.message.includes('Timed out')) {
+        console.warn(
+          chalk.yellow('Contract deployment exceeding configured timeout'),
+          error,
+        );
+      } else {
+        console.error(chalk.red('Contract deployment failed'), error);
+      }
+    }
+  }
+}
+
+async function baseDeploy<
```

Comment:
I'm a little spooked about the "duplication" instead of using the `deploy` function directly on the deployer -- I could see it not being obvious that this function is the main driver for some types of deployers, resulting in us missing out on changes to the .deploy function on the deployers.

wdyt? do you think this is unlikely to be an issue?

Reply:
i don't see this becoming an issue tbh, I would be maybe a little worried if we we had our own version of a router deployer's `deploy` function, but there's some categories of deployers (like the ones we listed here) where the `deploy` is simply just a loop across chains in the config and I don't think it's likely this will ever really change

Also in some ways this actually helps with any future migration to the new evm modules because they don't have a high-level `deploy` entrypoint. the `Module.Create()` entrypoint is analogous to `HyperlaneDeployer.deployContracts(chain)` we're leveraging here